src/input-range-select/input-range-select.tsx

```diff
@@ -151,41 +151,35 @@ export const InputRangeSelect = <T, V>({
     // =============================================================================
     // HELPER FUNCTION
     // =============================================================================
-    const getDisplayValue = (rangeType: RangeType): string | V => {
+    const getDisplayValue = (rangeType: RangeType): string => {
         const selected =
             rangeType === "from" ? selectedFromValue : selectedToValue;
+
+        if (!selected) return "";
+
         if (displayValueExtractor) {
             return displayValueExtractor(selected);
         }
 
         if (valueExtractor) {
-            return valueExtractor(selected);
+            const value = valueExtractor(selected);
+            return valueToStringFunction
+                ? valueToStringFunction(value)
+                : value?.toString() ?? "";
         }
-        return selected?.toString();
-    };
 
-    const convertValueToString = (value: V | string): string => {
-        if (typeof value === "string") {
-            return value;
-        } else {
-            return valueToStringFunction(value) || value.toString();
-        }
+        return selected.toString();
     };
 
-    const truncateValue = (type: RangeType, value: string | V) => {
+    const truncateValue = (type: RangeType, value: string) => {
         if (optionTruncationType === "middle") {
             let widthOfElement = 0;
             if (labelContainerRef[type] && labelContainerRef[type].current) {
                 widthOfElement =
                     labelContainerRef[type].current.getBoundingClientRect()
                         .width;
             }
-            return StringHelper.truncateOneLine(
-                convertValueToString(value),
```

Comment:
originally, the custom stringify would only be called when truncation type is middle